src/mono/mono/mini/aot-compiler.c

```diff
@@ -12909,8 +12909,15 @@ resolve_class (ClassProfileData *cdata)
 	}
 	if (cdata->inst) {
 		resolve_ginst (cdata->inst);
-		if (!cdata->inst->inst)
+		if (!cdata->inst->inst) {
+			/*
+			 * The instance might not be found if its arguments are in another assembly,
+			 * use the definition instead.
```

Comment:
How does this happen?   If an assembly is loaded at runtime and then an instance is created, for example?

Reply:
The profile might include methods like List<AClass>, and when corlib is AOT-ed, AClass-es assembly is not loaded.